nuxeo-services/login/nuxeo-platform-login-token-rest-api/src/test/java/org/nuxeo/ecm/restapi/test/AuthenticationTokensTest.java

```diff
@@ -0,0 +1,154 @@
+import javax.inject.Inject;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(FeaturesRunner.class)
+@Features({ RestServerFeature.class })
+@Jetty(port = 18090)
+@Deploy({ "org.nuxeo.ecm.platform.login.token", "org.nuxeo.ecm.platform.restapi.server.login.tokenauth"})
+public class AuthenticationTokensTest extends BaseTest {
+
+    @Inject
+    TokenAuthenticationService tokenAuthenticationService;
+
+    @Test
+    public void itCanQueryTokens() throws Exception {
+        // Check empty token list
+        ClientResponse response = getResponse(RequestType.GET, "/token");
+        assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+        JsonNode node = mapper.readTree(response.getEntityInputStream());
+        assertEquals(0, getEntries("tokens", node).size());
+
+        // acquire some tokens
+        String token1 = tokenAuthenticationService.acquireToken("Administrator", "app1", "device1", "", "rw");
+        String token2 = tokenAuthenticationService.acquireToken("Administrator", "app2", "device2", "", "rw");
+
+        nextTransaction();
+
+        // query tokens for current user
+        List<JsonNode> tokens = getTokens();
+        assertEquals(2, tokens.size());
+        assertEquals(token2, tokens.get(0).get("id").getTextValue());
+        assertEquals(token1, tokens.get(1).get("id").getTextValue());
+
+        // filter tokens by application
+        tokens = getTokens("app1");
+        assertEquals(1, tokens.size());
+        assertEquals(token1, tokens.get(0).get("id").getTextValue());
+    }
+
+    @Test
+    public void itCanRevokeTokens() throws Exception {
+        // acquire a token
+        String token1 = tokenAuthenticationService.acquireToken("Administrator", "app1", "device1", "", "rw");
+        nextTransaction();
+
+        // delete it
+        ClientResponse response = getResponse(RequestType.DELETE, "/token/" + token1);
+        assertEquals(Response.Status.NO_CONTENT.getStatusCode(), response.getStatus());
+
+        // check no tokens
+        List<JsonNode> tokens = getTokens();
+        assertEquals(0, tokens.size());
+    }
+
+    @Test
+    public void itCanCreateTokens() throws Exception {
+        // acquire a token
+        MultivaluedMap<String, String> params = new MultivaluedMapImpl();
+        params.put("application", Collections.singletonList("app"));
+        params.put("deviceId", Collections.singletonList("device"));
+        params.put("permission", Collections.singletonList("rw"));
+        ClientResponse response = getResponse(RequestType.POST, "/token", null, params, null, null);
+        assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
```

Comment:
Shouldn't POST returns `201` when creating a token (and `200` when just getting an already created token)?